modules/ctl/src/main/scala/hi/deploy.scala

```diff
@@ -11,13 +11,14 @@ object Deploy {
 
   final case class Deployment(gem: Container, postgres: Container)
 
-  private def destroyDeployment: CtlIO[Unit] =
+  private def destroyDeployment(newGemImage: Image): CtlIO[Unit] =
     gosub("Destroying current deployment, if any.") {
-      findRunningContainersWithLabel("gem.version").flatMap {
-        case Nil => info("None found, nothing to do!")
-        case cs  => cs.traverse { c =>
-          info(s"Stopping and removing ${c.hash}.") *> destroyContainer(c)
-        }.void
+      currentDeployment.flatMap {
+        case None => info("None found, nothing to do!")
+        case Some(Deployment(g, p)) =>
+          containerImage(g).flatMap { oldGemImage =>
+            destroyContainer(g, !(oldGemImage like newGemImage)) *> destroyContainer(p, false)
```

Comment:
When we shut down the existing deployment (containers `g` for Gem and `p` for Postgres) we also destroy `g`'s *image* if we're deploying a new version. Otherwise we just shut it down and the existing image will be re-used.